plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -194,11 +198,32 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
       };
     }
 
-    const dbResponse = rows.map(e => JSON.parse(e.final_entity!));
+    let entities: Entity[] = rows.map(e => JSON.parse(e.final_entity!));
+
+    if (request?.fields) {
+      entities = entities.map(e => request.fields!(e));
+    }
 
-    const entities = dbResponse.map(e =>
-      request?.fields ? request.fields(e) : e,
-    );
+    // TODO(freben): This is added as a compatibility guarantee, until we can be
```

Comment:
Can we make sure that this is happening for entities that never change? I mean my old `tractor` service that is not actively developed but very important. There might be a year until I change that and get a new Etag so that it restitches.
Can we do this with a database migration instead?

Reply:
@jhaals So I wracked my brain on what would be the safest, most robust path forward. I did start writing a migration loop that does the json update in-place, but realizing that on large catalogs it'll be a significant risk - the update will take quite some time unless we do some clever batching, blocking service startup time and increasing the likelihood that in-flight stitches will overlap. In the end I stepped back and instead marked everything for reprocessing, so that it'll happen in a distributed fashion. Made sure to wipe the output hashes so we know that the stitch actually will be performed. In a way, it is also the simpler option with the least amount of code. That also means that the compensation code that retrofits the ref when it's missing, still is in here. But one or two releases later we should be able to remove it. Give it another look, see if you think it makes sense.